distros/N/Net-Hulu/docs/Hulu.man
.\" Automatically generated by Pod::Man 2.22 (Pod::Simple 3.07)
.\"
.\" Standard preamble:
.\" ========================================================================
.de Sp \" Vertical space (when we can't use .PP)
.if t .sp .5v
.if n .sp
..
.de Vb \" Begin verbatim text
.ft CW
.nf
.ne \\$1
..
.de Ve \" End verbatim text
.ft R
.fi
..
.\" Set up some character translations and predefined strings.  \*(-- will
.\" give an unbreakable dash, \*(PI will give pi, \*(L" will give a left
.\" double quote, and \*(R" will give a right double quote.  \*(C+ will
.\" give a nicer C++.  Capital omega is used to do unbreakable dashes and
.\" therefore won't be available.  \*(C` and \*(C' expand to `' in nroff,
.\" nothing in troff, for use with C<>.
.tr \(*W-
.ds C+ C\v'-.1v'\h'-1p'\s-2+\h'-1p'+\s0\v'.1v'\h'-1p'
.ie n \{\
.    ds -- \(*W-
.    ds PI pi
.    if (\n(.H=4u)&(1m=24u) .ds -- \(*W\h'-12u'\(*W\h'-12u'-\" diablo 10 pitch
.    if (\n(.H=4u)&(1m=20u) .ds -- \(*W\h'-12u'\(*W\h'-8u'-\"  diablo 12 pitch
.    ds L" ""
.    ds R" ""
.    ds C` ""
.    ds C' ""
'br\}
.el\{\
.    ds -- \|\(em\|
.    ds PI \(*p
.    ds L" ``
.    ds R" ''
'br\}
.\"
.\" Escape single quotes in literal strings from groff's Unicode transform.
.ie \n(.g .ds Aq \(aq
.el       .ds Aq '
.\"
.\" If the F register is turned on, we'll generate index entries on stderr for
.\" titles (.TH), headers (.SH), subsections (.SS), items (.Ip), and index
.\" entries marked with X<> in POD.  Of course, you'll have to process the
.\" output yourself in some meaningful fashion.
.ie \nF \{\
.    de IX
.    tm Index:\\$1\t\\n%\t"\\$2"
..
.    nr % 0
.    rr F
.\}
.el \{\
.    de IX
..
.\}
.\"
.\" Accent mark definitions (@(#)ms.acc 1.5 88/02/08 SMI; from UCB 4.2).
.\" Fear.  Run.  Save yourself.  No user-serviceable parts.
.    \" fudge factors for nroff and troff
.if n \{\
.    ds #H 0
.    ds #V .8m
.    ds #F .3m
.    ds #[ \f1
.    ds #] \fP
.\}
.if t \{\
.    ds #H ((1u-(\\\\n(.fu%2u))*.13m)
.    ds #V .6m
.    ds #F 0
.    ds #[ \&
.    ds #] \&
.\}
.    \" simple accents for nroff and troff
.if n \{\
.    ds ' \&
.    ds ` \&
.    ds ^ \&
.    ds , \&
.    ds ~ ~
.    ds /
.\}
.if t \{\
.    ds ' \\k:\h'-(\\n(.wu*8/10-\*(#H)'\'\h"|\\n:u"
.    ds ` \\k:\h'-(\\n(.wu*8/10-\*(#H)'\`\h'|\\n:u'
.    ds ^ \\k:\h'-(\\n(.wu*10/11-\*(#H)'^\h'|\\n:u'
.    ds , \\k:\h'-(\\n(.wu*8/10)',\h'|\\n:u'
.    ds ~ \\k:\h'-(\\n(.wu-\*(#H-.1m)'~\h'|\\n:u'
.    ds / \\k:\h'-(\\n(.wu*8/10-\*(#H)'\z\(sl\h'|\\n:u'
.\}
.    \" troff and (daisy-wheel) nroff accents
.ds : \\k:\h'-(\\n(.wu*8/10-\*(#H+.1m+\*(#F)'\v'-\*(#V'\z.\h'.2m+\*(#F'.\h'|\\n:u'\v'\*(#V'
.ds 8 \h'\*(#H'\(*b\h'-\*(#H'
.ds o \\k:\h'-(\\n(.wu+\w'\(de'u-\*(#H)/2u'\v'-.3n'\*(#[\z\(de\v'.3n'\h'|\\n:u'\*(#]
.ds d- \h'\*(#H'\(pd\h'-\w'~'u'\v'-.25m'\f2\(hy\fP\v'.25m'\h'-\*(#H'
.ds D- D\\k:\h'-\w'D'u'\v'-.11m'\z\(hy\v'.11m'\h'|\\n:u'
.ds th \*(#[\v'.3m'\s+1I\s-1\v'-.3m'\h'-(\w'I'u*2/3)'\s-1o\s+1\*(#]
.ds Th \*(#[\s+2I\s-2\h'-\w'I'u*3/5'\v'-.3m'o\v'.3m'\*(#]
.ds ae a\h'-(\w'a'u*4/10)'e
.ds Ae A\h'-(\w'A'u*4/10)'E
.    \" corrections for vroff
.if v .ds ~ \\k:\h'-(\\n(.wu*9/10-\*(#H)'\s-2\u~\d\s+2\h'|\\n:u'
.if v .ds ^ \\k:\h'-(\\n(.wu*10/11-\*(#H)'\v'-.4m'^\v'.4m'\h'|\\n:u'
.    \" for low resolution devices (crt and lpr)
.if \n(.H>23 .if \n(.V>19 \
\{\
.    ds : e
.    ds 8 ss
.    ds o a
.    ds d- d\h'-1'\(ga
.    ds D- D\h'-1'\(hy
.    ds th \o'bp'
.    ds Th \o'LP'
.    ds ae ae
.    ds Ae AE
.\}
.rm #[ #] #H #V #F C
.\" ========================================================================
.\"
.IX Title "HULU 1"
.TH HULU 1 "2010-01-08" "perl v5.10.1" "User Contributed Perl Documentation"
.\" For nroff, turn off justification.  Always turn off hyphenation; it makes
.\" way too many mistakes in technical documents.
.if n .ad l
.nh
.SH "NAME"
Net::Hulu \- Perl extension for the Net::Hulu name space.  Can be use to download
and process all XML\-formatted RSS feeds provided by the popular site hulu.com.
.SH "ABSTRACT"
.IX Header "ABSTRACT"
Net::Hulu will download and process XML-formatted \s-1RSS\s0 feeds provided by hulu.com.
.SH "SYNOPSIS"
.IX Header "SYNOPSIS"
.Vb 11
\&  use strict;
\&  use warnings;
\&  use Net::Hulu qw(
\&                get_recent_videos get_recent_shows get_recent_movies get_highest_rated_videos
\&                get_popular_videos_today get_popular_videos_this_week get_popular_videos_all_time
\&                get_soon_to_expire_videos get_recent_blog_postings download_recent_videos_xml
\&                download_recent_shows_xml download_recent_movies_xml download_highest_rated_videos_xml
\&                download_popular_videos_today_xml download_popular_videos_this_week_xml
\&                download_popular_videos_all_time_xml download_soon_to_expire_videos_xml
\&                download_recent_blog_postings_xml get_popular_videos_this_month
\&                download_popular_videos_this_month_xml);
.Ve
.SH "DESCRIPTION"
.IX Header "DESCRIPTION"
Net::Hulu provides methods to download all XML-formatted \s-1RSS\s0 feeds provided by hulu.com and 
then return a data structure to the user containing the Data from the \s-1RSS\s0 feeds.
.PP
The following methods will download \s-1RSS\s0 feeds from hulu.com to directory where Hulu.pm is installed.
Method download_recent_videos_xml will download \s-1RSS\s0 feed for recent videos.
Method download_recent_shows_xml will download \s-1RSS\s0 feed for recent shows.
Method download_recent_movies_xml will download \s-1RSS\s0 feed for recent movies.
Method download_highest_rated_videos_xml will download \s-1RSS\s0 feed for highest rated videos.
Method download_popular_videos_today_xml will download \s-1RSS\s0 feed for today's popular videos.
Method download_popular_videos_this_week_xml will download \s-1RSS\s0 feed for this week's popular videos.
Method download_popular_videos_this_month_xml will download \s-1RSS\s0 feed for this month's popular videos.
Method download_popular_videos_all_time_xml will download \s-1RSS\s0 feed for most popular videos of all time.
Method download_soon_to_expire_videos_xml will download \s-1RSS\s0 feed for soon to expire videos.
Method download_recent_blog_postings_xml will download \s-1RSS\s0 feed for recent blog postings to hulu.com.
.PP
The following methods will process all \s-1RSS\s0 feeds provided by hulu.com.
Method get_recent_videos returns a list of recently added videos to the user.
Method get_recent_shows returns a list of recently added shows to the user.
Method get_recent_movies returns a list of recently added movies to the user.
Method get_highest_rated_videos returns a list of highest rated videos to the user.
Method get_popular_videos_today returns a list of most popular videos today to the user.
Method get_popular_videos_this_week returns a list of most popular videos this week to the user.
Method get_popular_videos_this_month returns a list of most popular videos this month to the user.
Nethod get_popular_videos_all_time returns a list of most popular videos of all time to the user.
Method get_soon_to_expire_videos returns a list of soon to expire videos to the user.
Method get_recent_blog_postings returns recent blog postings for hulu.com to the user.
.SS "Getting Started"
.IX Subsection "Getting Started"
The following code will download all the XML-formatted \s-1RSS\s0 feeds provided by hulu.com,
process their content, and print the results for the user.
.PP
.Vb 10
\&   use strict;
\&   use warnings;
\&   use Net::Hulu qw(
\&                   get_recent_videos get_recent_shows get_recent_movies get_highest_rated_videos
\&                   get_popular_videos_today get_popular_videos_this_week get_popular_videos_all_time
\&                   get_soon_to_expire_videos get_recent_blog_postings download_recent_videos_xml
\&                   download_recent_shows_xml download_recent_movies_xml download_highest_rated_videos_xml
\&                   download_popular_videos_today_xml download_popular_videos_this_week_xml
\&                   download_popular_videos_all_time_xml download_soon_to_expire_videos_xml
\&                   download_recent_blog_postings_xml);
\&
\&   download_recent_videos_xml();
\&   get_recent_videos();
\&
\&   download_recent_shows_xml();
\&   get_recent_shows();
\&
\&   download_recent_movies_xml();
\&   get_recent_movies();
\&
\&   download_highest_rated_videos_xml();
\&   get_highest_rated_videos();
\&
\&   download_popular_videos_today_xml();
\&   get_popular_videos_today();
\&
\&   download_popular_videos_this_week_xml();
\&   get_popular_videos_this_week();
\&
\&   download_popular_videos_this_month_xml();
\&   get_popular_videos_this_month();
\&
\&   download_popular_videos_all_time_xml();
\&   get_popular_videos_all_time();
\&
\&   download_soon_to_expire_videos_xml();
\&   get_soon_to_expire_videos();
\&
\&   download_recent_blog_postings_xml();
\&   get_recent_blog_postings();
.Ve
.SS "download_recent_videos_xml"
.IX Subsection "download_recent_videos_xml"
The subroutine download_recent_videos_xml will download http://rss.hulu.com/HuluRecentlyAddedVideos?format=xml
to the directory where Hulu.pm is installed.
.SS "get_recent_videos"
.IX Subsection "get_recent_videos"
The subroutine get_recent_videos will return a list of recently added videos for the user.
.SS "download_recent_shows_xml"
.IX Subsection "download_recent_shows_xml"
The subroutine download_recent_shows_xml will download http://rss.hulu.com/HuluRecentlyAddedShows?format=xml
to the directory where Hulu.pm is installed.
.SS "get_recent_shows"
.IX Subsection "get_recent_shows"
The subroutine get_recent_shows will return a list of recently added shows for the user.
.SS "download_recent_movies_xml"
.IX Subsection "download_recent_movies_xml"
The subroutine download_recent_movies_xml will download http://rss.hulu.com/HuluRecentlyAddedMovies?format=xml
to the directory where Hulu.pm is installed.
.SS "get_recent_movies"
.IX Subsection "get_recent_movies"
The subroutine get_recent_movies will return a list of recently added movies for the user.
.SS "download_highest_rated_videos_xml"
.IX Subsection "download_highest_rated_videos_xml"
The subroutine download_highest_rated_videos_xml will download http://www.hulu.com/feed/highest_rated/videos
to the directory where Hulu.pm is installed.
.SS "get_highest_rated_videos"
.IX Subsection "get_highest_rated_videos"
The subroutine get_highest_rated_videso will return a list of the highest rated videos for the user.
.SS "download_popular_videos_today_xml"
.IX Subsection "download_popular_videos_today_xml"
The subroutine download_popular_videos_today_xml will download http://rss.hulu.com/HuluPopularVideosToday?format=xml
to the directory where Hulu.pm is installed.
.SS "get_popular_videos_today"
.IX Subsection "get_popular_videos_today"
The subroutine get_popular_videos_today will return a list of the most popular videos today for the user.
.SS "download_popular_videos_this_week_xml"
.IX Subsection "download_popular_videos_this_week_xml"
The subroutine download_popular_videos_this_week_xml will download http://rss.hulu.com/HuluPopularVideosThisWeek?format=xml
to the directory where Hulu.pm is installed.
.SS "get_popular_videos_this_week"
.IX Subsection "get_popular_videos_this_week"
The subroutine get_popular_videos_this_week will return a list of most popular videos this week for the user.
.SS "download_popular_videos_this_month_xml"
.IX Subsection "download_popular_videos_this_month_xml"
The subroutine get_popular_videos_this_month_xml will download http://rss.hulu.com/HuluPopularVideosThisMonth?format=xml
to the directory where Hulu.pm is installed.
.SS "get_popular_videos_this_month"
.IX Subsection "get_popular_videos_this_month"
The subroutine get_popular_videos_this_month will return a list of most popular videos this month for the user.
.SS "download_popular_videos_all_time_xml"
.IX Subsection "download_popular_videos_all_time_xml"
The subroutine download_popular_videos_all_time_xml will download http://rss.hulu.com/HuluPopularVideosAllTime?format=xml
to the directory where Hulu.pm is installed.
.SS "get_popular_videos_all_time"
.IX Subsection "get_popular_videos_all_time"
The subroutine get_popular_videos_all_time will return a list of most popular videos of all time to the user.
.SS "download_soon_to_expire_videos_xml"
.IX Subsection "download_soon_to_expire_videos_xml"
The subroutine download_soon_to_expire_videos_xml will download http://www.hulu.com/feed/expiring/videos
to the directory where Hulu.pm is installed.
.SS "get_soon_to_expire_videos"

distros/N/Net-Hulu/lib/Net/Hulu.pm
#################################################################################
#########This Perl module represents the CPAN name space Net::Hulu.##############
####################Written by Gerald L. Hevener, M.S.###########################
##############AKA: jackl0phty in the whitehat hacker community.##################
#########This module is licensed under the same terms as Perl itself.############
#########Maintainer's Email:  hevenerg {[AT]} marshall {[DOT]} edu.##############
#After years of using free (as in beer) software, thought I'd try to give back. #
#################################################################################

# declare package name
package Net::Hulu;

use 5.006000;
use strict;
use warnings;
use Carp;
use XML::Twig;
use LWP::Simple;

require Exporter;

our @ISA = qw(Exporter);

# Items to export into callers namespace by default. Note: do not export
# names by default without a very good reason. Use EXPORT_OK instead.
# Do not simply export all your public functions/methods/constants.

# This allows declaration	use Net::Hulu ':all';
# If you do not need this, moving things directly into @EXPORT or @EXPORT_OK
# will save memory.
our %EXPORT_TAGS = ( 'all' => [ qw(
	
) ] );

our @EXPORT_OK = ( @{ $EXPORT_TAGS{'all'} } );

our @EXPORT = qw(
		get_recent_videos get_recent_shows get_recent_movies get_highest_rated_videos
                get_popular_videos_today get_popular_videos_this_week
		get_popular_videos_this_month get_popular_videos_all_time
                get_soon_to_expire_videos get_recent_blog_postings download_recent_videos_xml
                download_recent_shows_xml download_recent_movies_xml download_highest_rated_videos_xml
                download_popular_videos_today_xml download_popular_videos_this_week_xml
		download_popular_videos_this_month_xml download_popular_videos_all_time_xml
		download_soon_to_expire_videos_xml download_recent_blog_postings_xml
);

our $VERSION = '0.03';

# declare variables for recent videos
my $recent_videos_url = "http://rss.hulu.com/HuluRecentlyAddedVideos?format=xml";
my $recent_videos_root;
my $recent_videos_xml;
my $recent_videos_twig = new XML::Twig(TwigRoots => {'item' => 1, pretty_print => 'indented'});

# declare variables for recent shows
my $recent_shows_url = "http://rss.hulu.com/HuluRecentlyAddedShows?format=xml";
my $recent_shows_root;
my $recent_shows_xml;
my $recent_shows_twig = new XML::Twig(TwigRoots => {'item' => 1, pretty_print => 'indented'});

# declare variables for recent movies
my $recent_movies_url = "http://rss.hulu.com/HuluRecentlyAddedMovies?format=xml";
my $recent_movies_root;
my $recent_movies_xml;
my $recent_movies_twig = new XML::Twig(TwigRoots => {'item' => 1, pretty_print => 'indented'});

# declare variables for highest rated videos
my $highest_rated_videos_url = "http://www.hulu.com/feed/highest_rated/videos";
my $highest_rated_videos_root;
my $highest_rated_videos_xml;
my $highest_rated_videos_twig = new XML::Twig(TwigRoots => {'item' => 1, pretty_print => 'indented'});

# declare variables for popular videos today
my $popular_videos_today_url = "http://rss.hulu.com/HuluPopularVideosToday?format=xml";
my $popular_videos_today_root;
my $popular_videos_today_xml;
my $popular_videos_today_twig = new XML::Twig(TwigRoots => {'item' => 1, pretty_print => 'indented'});

# declare variables for popular videos this week
my $popular_videos_this_week_url = "http://rss.hulu.com/HuluPopularVideosThisWeek?format=xml";
my $popular_videos_this_week_root;
my $popular_videos_this_week_xml;
my $popular_videos_this_week_twig = new XML::Twig(TwigRoots => {'item' => 1, pretty_print => 'indented'});

# declare variables for popular videos this month
my $popular_videos_this_month_url = "http://rss.hulu.com/HuluPopularVideosThisMonth?format=xml";
my $popular_videos_this_month_root;
my $popular_videos_this_month_xml;
my $popular_videos_this_month_twig = new XML::Twig(TwigRoots => {'item' => 1, pretty_print => 'indented'});

# declare variables for popular videos of all time
my $popular_videos_all_time_url = "http://rss.hulu.com/HuluPopularVideosAllTime?format=xml";
my $popular_videos_all_time_root;
my $popular_videos_all_time_xml;
my $popular_videos_all_time_twig = new XML::Twig(TwigRoots => {'item' => 1, pretty_print => 'indented'});

# declare variables for soon to expire videos 
my $soon_to_expire_videos_url = "http://www.hulu.com/feed/expiring/videos";
my $soon_to_expire_videos_root;
my $soon_to_expire_videos_xml;
my $soon_to_expire_videos_twig = new XML::Twig(TwigRoots => {'item' => 1, pretty_print => 'indented'});

# declare variables for recent blog postings
my $recent_blog_postings_url = "http://rss.hulu.com/HuluBlog?format=xml";
my $recent_blog_postings_root;
my $recent_blog_postings_xml;
my $recent_blog_postings_twig = new XML::Twig(TwigRoots => {'item' => 1, pretty_print => 'indented'});

# preloaded methods go here.
######################Begin Primary Subroutines##########################

sub get_recent_videos {

	#Turn on strict and warnings.
	use strict;
	use warnings;

        $recent_videos_twig->parsefile("HuluRecentlyAddedVideos?format=xml");

        #set root of the twig (channel).
        $recent_videos_root = $recent_videos_twig->root;

        #get recent videos titles.
        foreach my $recent_videos_titles ($recent_videos_root->children('item')) {

                print $recent_videos_titles->first_child_text('title');
                print "\n";
        }

# sub get_recent_videos()
}

sub get_recent_shows {

	#Turn on strict and warnings.
	use strict;
	use warnings;

        #parse xml file for recent shows.
        $recent_shows_twig->parsefile("HuluRecentlyAddedShows?format=xml");

        #set root of the twig (channel).
        $recent_shows_root = $recent_shows_twig->root;

        #get recent videos titles.
        foreach my $recent_shows_titles ($recent_shows_root->children('item')) {

                print $recent_shows_titles->first_child_text('title');
                print "\n";
        }

# sub get_recent_shows()
}

sub get_recent_movies {

	#Turn on strict and warnings.
	use strict;
	use warnings;

        #parse xml file for recent movies.
        $recent_movies_twig->parsefile("HuluRecentlyAddedMovies?format=xml");

        #set root of the twig (channel).
        $recent_movies_root = $recent_movies_twig->root;

        #get recent movies titles.
        foreach my $recent_movies_titles ($recent_movies_root->children('item')) {

                print $recent_movies_titles->first_child_text('title');
                print "\n";
        }

# sub get_recent_movies() 
}

sub get_highest_rated_videos {

	#Turn on strict and warnings.
	use strict;
	use warnings;

        #parse xml file for highest rated videos.
        $highest_rated_videos_twig->parsefile("videos");

        #set root of the twig (channel).
        $highest_rated_videos_root = $highest_rated_videos_twig->root;

        #get highest rated videos.
        foreach my $highest_rated_videos_titles ($highest_rated_videos_root->children('item')) {

                print $highest_rated_videos_titles->first_child_text('title');
                print "\n";
        }

# sub get_highest_rated_videos()
}

sub get_popular_videos_today {

	#Turn on strict and warnings.
	use strict;
	use warnings;

        #parse xml file for popular videos today.
        $popular_videos_today_twig->parsefile("HuluPopularVideosToday?format=xml");

        #set root of the twig (channel).
        $popular_videos_today_root = $popular_videos_today_twig->root;

        #get popular videos today.
        foreach my $popular_videos_today_titles ($popular_videos_today_root->children('item')) {

                print $popular_videos_today_titles->first_child_text('title');
                print "\n";
        }

# sub get_popular_videos_today()
}

sub get_popular_videos_this_week {

	#Turn on strict and warnings.
	use strict;
	use warnings;

        #parse xml file for popular videos this week.
        $popular_videos_this_week_twig->parsefile("HuluPopularVideosThisWeek?format=xml");

        #set root of the twig (channel).
        $popular_videos_this_week_root = $popular_videos_this_week_twig->root;

        #get popular videos this week.
        foreach my $popular_videos_this_week_titles ($popular_videos_this_week_root->children('item')) {

                print $popular_videos_this_week_titles->first_child_text('title');
                print "\n";
        }

# sub get_popular_videos_this_week()
}

sub get_popular_videos_this_month {

	#Turn on strict and warnings.
        use strict;
        use warnings;

        #parse xml file for popular videos this month.
        $popular_videos_this_month_twig->parsefile("HuluPopularVideosThisMonth?format=xml");

        #set root of the twig (channel).
        $popular_videos_this_month_root = $popular_videos_this_month_twig->root;

        #get popular videos this month.
        foreach my $popular_videos_this_month_titles ($popular_videos_this_month_root->children('item')) {

                print $popular_videos_this_month_titles->first_child_text('title');
                print "\n";
        }

# sub get_popular_videos_this_month()
}

sub get_popular_videos_all_time {
	
	#Turn on strict and warnings.
	use strict;
	use warnings;

        #parse xml file for popular videos of all time.
        $popular_videos_all_time_twig->parsefile("HuluPopularVideosAllTime?format=xml");

        #set root of the twig (channel)
        $popular_videos_all_time_root = $popular_videos_all_time_twig->root;

        #get popular videos all time
        foreach my $popular_videos_all_time_titles ($popular_videos_all_time_root->children('item')) {

                print $popular_videos_all_time_titles->first_child_text('title');
                print "\n";
        }

# sub get_popular_videos_all_time()         
}

sub get_soon_to_expire_videos {

	#Turn on strict and warnings.
	use strict;
	use warnings;

        #parse xml file for soon to expire videos.
        $soon_to_expire_videos_twig->parsefile("videos.1");

        #set root of the twig (channel).
        $soon_to_expire_videos_root = $soon_to_expire_videos_twig->root;

        #get soon to expire videos.
        foreach my $soon_to_expire_videos_titles ($soon_to_expire_videos_root->children('item')) {

                print $soon_to_expire_videos_titles->first_child_text('title');
                print "\n";
        }

# sub get_soon_to_expire_videos()
}

sub get_recent_blog_postings {

	#Turn on strict and warnings.
	use strict;
	use warnings;

        #parse xml file for hulu's recent blog postings. (Hulu's primary RSS feed).
        $recent_blog_postings_twig->parsefile("HuluBlog?format=xml");

        #set root of the twig (channel).
        $recent_blog_postings_root = $recent_blog_postings_twig->root;

        #get hulu's recent blog postings
        foreach my $recent_blog_postings_titles ($recent_blog_postings_root->children('item')) {

                print $recent_blog_postings_titles->first_child_text('title');
                print "\n";
        }

# sub get_recent_blog_postings()         
}

######################End of primary subroutines##########################

########Begin subroutines that download XML RSS feeds from Hulu###########

sub download_recent_videos_xml {

	#Turn on strict and warnings.
	use strict;
	use warnings;

        #get xml using LWP::Simple.
        $recent_videos_xml = get $recent_videos_url;

        #get rid of non-ascii chars.
        $recent_videos_xml =~ s/[^[:ascii:]]+//g;

        #save XML to file.
        open( my $RECENT_VIDEOS_FH, ">", "HuluRecentlyAddedVideos?format=xml" ) or confess "Can't open file: $!";

                #print recent videos to file in PWD.
                print $RECENT_VIDEOS_FH "$recent_videos_xml";

        close($RECENT_VIDEOS_FH);

# sub download_recent_videos_xml()
}

sub download_recent_shows_xml {

	#Turn on strict and warnings.
	use strict;
	use warnings;

        #get xml using LWP::Simple.
        $recent_shows_xml = get $recent_shows_url;

        #get rid of non-ascii chars.
        $recent_shows_xml =~ s/[^[:ascii:]]+//g;

        #save XML to file.
        open( my $RECENT_SHOWS_FH, ">", "HuluRecentlyAddedShows?format=xml" ) or confess "Can't open file: $!";

                #print recent shows to file in PWD.
                print $RECENT_SHOWS_FH "$recent_shows_xml";

        close($RECENT_SHOWS_FH);

# sub download_recent_shows_xml()
}

sub download_recent_movies_xml {

	#Turn on strict and warnings.
	use strict;
	use warnings;

        #get xml using LWP::Simple.
        $recent_movies_xml = get $recent_movies_url;

        #get rid of non-ascii chars.
        $recent_movies_xml =~ s/[^[:ascii:]]+//g;

        #save XML to file.
        open( my $RECENT_MOVIES_FH, ">", "HuluRecentlyAddedMovies?format=xml" ) or confess "Can't open file: $!";

                #print recent movies to file in PWD.
                print $RECENT_MOVIES_FH "$recent_movies_xml";

        close($RECENT_MOVIES_FH);

# sub download_recent_movies().
}

sub download_highest_rated_videos_xml {

	#Turn on strict and warnings.
	use strict;
	use warnings;

        #get xml using LWP::Simple.
        $highest_rated_videos_xml = get $highest_rated_videos_url;

        #get rid of non-ascii chars.
        $highest_rated_videos_xml =~ s/[^[:ascii:]]+//g;

        #save XML to file.
        open( my $HIGHEST_RATED_VIDEOS_FH, ">", "videos" ) or confess "Can't open file: $!";

                #print highest rated videos to file in PWD.
                print $HIGHEST_RATED_VIDEOS_FH "$highest_rated_videos_xml";

        close($HIGHEST_RATED_VIDEOS_FH);

# sub download_highest_rated_videos_xml().
}

sub download_popular_videos_today_xml {

	#Turn on strict and warnings.
	use strict;
	use warnings;

        #get xml using LWP::Simple.
        $popular_videos_today_xml = get $popular_videos_today_url;

        #get rid of non-ascii chars.
        $popular_videos_today_xml =~ s/[^[:ascii:]]+//g;

        #save XML to file.
        open( my $POPULAR_VIDEOS_TODAY_FH, ">", "HuluPopularVideosToday?format=xml" ) or confess "Can't open file: $!";

                #print popular videos for today to file in PWD.
                print $POPULAR_VIDEOS_TODAY_FH "$popular_videos_today_xml";

        close($POPULAR_VIDEOS_TODAY_FH);

# sub download_popular_videos_today_xml.
}

sub download_popular_videos_this_week_xml {

	#Turn on strict and warnings.
	use strict;
	use warnings;

        #get xml using LWP::Simple.
        $popular_videos_this_week_xml = get $popular_videos_this_week_url;

        #get rid of non-ascii chars.
        $popular_videos_this_week_xml =~ s/[^[:ascii:]]+//g;

        #save XML to file.
        open( my $POPULAR_VIDEOS_THIS_WEEK_FH, ">", "HuluPopularVideosThisWeek?format=xml" ) or confess "Can't open file: $!";

                #print popular videos for this week to file in PWD.
                print $POPULAR_VIDEOS_THIS_WEEK_FH "$popular_videos_this_week_xml";

        close($POPULAR_VIDEOS_THIS_WEEK_FH);

# sub download_popular_videos_this_week().
}

sub download_popular_videos_this_month_xml {

        #Turn on strict and warnings.
        use strict;
        use warnings;

        #get xml using LWP::Simple.
        $popular_videos_this_month_xml = get $popular_videos_this_month_url;

        #get rid of non-ascii chars.
        $popular_videos_this_month_xml =~ s/[^[:ascii:]]+//g;

        #save XML to file.
        open( my $POPULAR_VIDEOS_THIS_MONTH_FH, ">", "HuluPopularVideosThisMonth?format=xml" ) or confess "Can't open file: $!";

                #print popular videos for this month to file in PWD.
                print $POPULAR_VIDEOS_THIS_MONTH_FH "$popular_videos_this_month_xml";

        close($POPULAR_VIDEOS_THIS_MONTH_FH);

# sub download_popular_videos_this_month().
}

sub download_popular_videos_all_time_xml {

	#Turn on strict and warnings.
	use strict;
	use warnings;

        #get xml using LWP::Simple.
        $popular_videos_all_time_xml = get $popular_videos_all_time_url;

        #get rid of non-ascii chars.
        $popular_videos_all_time_xml =~ s/[^[:ascii:]]+//g;

        #save XML to file.
        open( my $POPULAR_VIDEOS_ALL_TIME_FH, ">", "HuluPopularVideosAllTime?format=xml" ) or confess "Can't open file: $!";

                #print popular videos of all time to file in PWD.
                print $POPULAR_VIDEOS_ALL_TIME_FH "$popular_videos_all_time_xml";

        close($POPULAR_VIDEOS_ALL_TIME_FH);

# sub download_popular_videos_all_time_xml()
}

sub download_soon_to_expire_videos_xml {

	#Turn on strict and warnings.
	use strict;
	use warnings;

        #get xml using LWP::Simple.
        $soon_to_expire_videos_xml = get $soon_to_expire_videos_url;

        #get rid of non-ascii chars.
        $soon_to_expire_videos_xml =~ s/[^[:ascii:]]+//g;

        #save XML to file.
        open( my $SOON_TO_EXPIRE_VIDEOS_FH, ">", "videos" ) or confess "Can't open file: $!";

                #print soon to expire videos to file in PWD.
                print $SOON_TO_EXPIRE_VIDEOS_FH "$soon_to_expire_videos_xml";

        close($SOON_TO_EXPIRE_VIDEOS_FH);

# sub download_soon_to_expire_videos_xml().
}

sub download_recent_blog_postings_xml {

	#Turn on strict and warnings.
	use strict;
	use warnings;

        #get xml using LWP::Simple.
        $recent_blog_postings_xml = get $recent_blog_postings_url;

        #get rid of non-ascii chars.
        $recent_blog_postings_xml =~ s/[^[:ascii:]]+//g;

        #save XML to file.
        open( my $RECENT_BLOG_POSTINGS_FH, ">", "HuluBlog?format=xml" ) or confess "Can't open file: $!";

                #print recent blog postings to file in PWD.
                print $RECENT_BLOG_POSTINGS_FH "$recent_blog_postings_xml";

        close($RECENT_BLOG_POSTINGS_FH);

# sub download_recent_blog_postings_xml().
}

# Modules must return a true value
1;
